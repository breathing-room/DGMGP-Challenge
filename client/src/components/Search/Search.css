.search-bar {
	padding-top: 15px;
	margin: 0 auto;
	display: inline-block
}

.search-input {
	width: 300px;
	padding: 8px;
	margin: 10px 5px;
	box-sizing: border-box; 
}

.search-btn {
	padding: 10px;
	margin: 0 auto;
	border-radius: 20%;
	background-color: aliceblue;
}