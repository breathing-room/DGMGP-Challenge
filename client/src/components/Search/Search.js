import React, { useState } from 'react';
import './Search.css'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSearch } from '@fortawesome/free-solid-svg-icons'
import PropTypes from 'prop-types'

const Search = ({ handleLocationSearch }) => {
	const [ query, setQuery ] = useState('');

	const handleChange = (e) => {
		setQuery(e.target.value);
	}

	const handleKeyPress = event => {
		if (event.key === 'Enter') {
			handleLocationSearch(query);
		}
	  };

	return (
		<div className='search-bar'> 
			<input 
				data-testid='input'
				type='text' 
				value={query} 
				placeholder='Search for a city' 
				className='search-input' 
				size='100' 
				onChange={handleChange} 
				onKeyPress={handleKeyPress}/>
			<button data-testid='search-btn' className='search-btn' onClick={() => { handleLocationSearch(query); setQuery('')}}>
				<FontAwesomeIcon icon={faSearch} style={{color:'#373838'}} />
			</button>
		</div>
	)
}

Search.propTypes = {
	query: PropTypes.string,
	handleLocationSearch: PropTypes.func
}

export default Search;