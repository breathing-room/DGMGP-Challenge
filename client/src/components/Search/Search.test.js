import React from 'react';
import Search from './Search';
import sinon from 'sinon';
import { getByTestId } from '@testing-library/dom'

import { mount, shallow } from 'enzyme';

describe('Search component', () => {
    let wrapper;
    const handleLocationSearch = jest.fn();
    const spy = sinon.spy()
    beforeEach(() => {
        wrapper = mount(<Search handleLocationSearch={handleLocationSearch} onButtonClick={spy}/>)
    })

    test('it renders correctly' , () => {
        const shallowWrap = shallow(
            <Search handleLocationSearch={handleLocationSearch}/>
        );
        expect(shallowWrap).toMatchSnapshot();

    })

    test('it renders a search bar', () => {
        expect(wrapper.find('#search-input'));
    })

    test('it renders a search button', () => {
       
        expect(wrapper.find('#search-btn'));
    })

    test('calls a function on search button click', () => {
        getByTestId('search-btn')
        .simulate('click');
        

        expect(spy.calledOnce).toBe(true);

        
    //     const onButtonClickMock = jest.fn();
    //     const wrapper = shallow(
    //         <button
    //         onButtonClick={onButtonClickMock}
    //         />,
    //         );
    //     const button = wrapper.find('button');
    //     button.simulate('click');
      
    //   expect(onButtonClickMock).toHaveBeenCalledTimes(1);
    //   expect(onButtonClickMock).toHaveBeenCalledWith(true);
    });

    // test('passes a search query to a submit function on button click', () => {
    //     const query = wrapper.find('input').props().value;

    // })

})
