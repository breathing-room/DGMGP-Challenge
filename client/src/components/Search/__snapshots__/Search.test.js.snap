// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component it renders correctly 1`] = `ShallowWrapper {}`;
